metadata-ingestion/src/datahub/ingestion/source/identity/azure_ad.py

```diff
@@ -270,16 +270,33 @@ def _get_azure_ad_data(self, kind: str) -> Iterable[List]:
                 self.report.report_failure("_get_azure_ad_data_", error_str)
                 continue
 
+    def _map_identity_to_urn(self, func, id_to_extract, mapping_identifier):
+        result, error_str = None, None
+        try:
+            result = func(id_to_extract)
+        except ValueError as e:
+            error_str = (
+                "Failed to extract DataHub {} from Azure AD User {} due to '{}'".format(
```

Comment:
Are we always mapping from an Azure AD user? Seems that we can map from groups also

Reply:
this got left out. will fix